presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -136,19 +136,18 @@ private QueryStateInfo getQueryStateInfo(
     @GET
     @Path("{queryId}")
     @Produces(MediaType.APPLICATION_JSON)
-    public void getQueryStateInfo(@PathParam("queryId") String queryId, @HeaderParam(X_FORWARDED_PROTO) String xForwardedProto,
+    public void getQueryStateInfo(@PathParam("queryId") QueryId queryId, @HeaderParam(X_FORWARDED_PROTO) String xForwardedProto,
```

Comment:
Inject query id as QueryId type instead of as String type and then parsing it into a QueryId immediately after

Reply:
Removed the change for this PR